src/order/chain.lean

```diff
@@ -181,7 +181,8 @@ begin
   case union : s hs ih
   { apply or.imp_left h.antisymm',
     apply classical.by_contradiction,
-    simp [not_or_distrib, sUnion_subset_iff, not_forall],
+    simp only [not_or_distrib, sUnion_subset_iff, not_forall, exists_prop, and_imp,
```

Comment:
What goes wrong without this change?

Reply:
(and is there another simp lemma we could add to fix it)

Reply:
I didn't check in detail, but it seems that a few lines later, the goal is not what was expected (`false` instead of something more complicated, IIRC). On [Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.60not_not.60.20and.20.60simp.60/near/289884490), I said that one could possibly simplify the proof, but I didn't look into this further.

Reply:
The `simp only` version gives `∀ (x : set α), x ∈ s → ¬x ⊆ c₁ → ¬succ_chain r c₁ ⊆ ⋃₀ s → false`; without `only`, we get `∀ (x : set α), x ∈ s → ¬x ⊆ c₁ → succ_chain r c₁ ⊆ ⋃₀ s`, and then it complains at `exact h₁ (subset_succ_chain.trans h)` that `term has type false but is expected to have type h₂ ∈ succ_chain r c₁ → h₂ ∈ ⋃₀ s`. One could add `exfalso` before the `exact` statement, and it would work, I think.

Reply:
But this gives further errors down the road.
The point is that the goal got (more) simplified, so the part of the proof following the `simp` needs to be adapted to that:
```
   intros c₃ hc₃ h₁,
    obtain h | h := chain_closure_succ_total_aux hc₁ (hs c₃ hc₃) (λ c₄, ih _ hc₃),
    { exfalso,
      exact h₁ (subset_succ_chain.trans h) },
    obtain h' | h' := ih c₃ hc₃ hc₁ h,
    { exfalso, exact h₁ h'.subset },
    { exact (h'.trans $ subset_sUnion_of_mem hc₃) } }
```

Reply:
Oh, I've just realized this is a non-terminal `simp`, so my concern is irrelevant.

Reply:
Does it still make sense to change the proof (adding `not_imp_false_iff` to the `simp` lemmas), or should I just leave it?